### ISSUE TITLE
OAuth auth token rejected, not clear why

### ISSUE THREAD

**Issue by MrCreosote** [kootenpv/yagmail#79]
```
$ python -c "import yagmail; yagmail.SMTP('[redacted]@gmailcom', oauth2_file='~/.[redacted]_gmail_oauth2_creds.json').send(subject='test done', contents='test whee')"
Your 'google_client_id': 
Your 'google_client_secret': 
Navigate to the following URL to auth:
https://accounts.google.com/o/oauth2/auth?client_id=[redacted].apps.googleusercontent.com&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob&response_type=code&scope=https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fgmail.send%2F
Enter verification code: [redacted]
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/local/lib/python2.7/dist-packages/yagmail/sender.py", line 113, in send
    return self._attempt_send(addresses['recipients'], msg.as_string())
  File "/usr/local/lib/python2.7/dist-packages/yagmail/sender.py", line 119, in _attempt_send
    result = self.smtp.sendmail(self.user, recipients, msg_string)
  File "/usr/lib/python2.7/smtplib.py", line 731, in sendmail
    raise SMTPSenderRefused(code, resp, from_addr)
smtplib.SMTPSenderRefused: (530, '5.5.1 Authentication Required. Learn more at\n5.5.1  https://support.google.com/mail/?p=WantAuthError 189sm11302089pgj.67 - gsmtp', '[redacted]@gmailcom')
```
Following the link (and trying numerous variants with and without whitespace) didn't provide any useful information as to why the verification code was rejected. I double checked the code on the last page of the Google OAuth2 flow against the terminal and it looks fine. Not sure where to go from here.

**Comment by kootenpv** [comment]
Did you fill in the client id and secret according to an app you created? I've been planning to write about how to create an app in google to get an id/secret. 
In case you haven't done this let me know.

Do you have this information in the oauth2 file?

    {
        "google_refresh_token": "........",
        "google_client_secret": "........",
        "google_client_id": ".........."
    }

**Comment by MrCreosote** [comment]
No problem. Thanks for taking the time to write and maintain yagmail. 

> Did you fill in the client id and secret according to an app you created? 
Yes, I suppose yagmail just shuts off the terminal echo.

> Do you have this information in the oauth2 file?
Yes.

**Comment by MrCreosote** [comment]
BTW, one minor issue I've noticed is that if you accidentally copy whitespace around the id/secret it'll also cause either bad urls or 401s when trying to get the refresh token. Adding a trim() in these lines would be helpful:

https://github.com/kootenpv/yagmail/blob/master/yagmail/oauth2.py#L110-L112

**Comment by kootenpv** [comment]
I [recorded a video](https://youtu.be/F_mPuKYfrcM) to show how it can be used with the latest version :)

Indeed, I will add the strip() to id and secret, it's a good suggestion!

**Comment by kootenpv** [comment]
Forgot to include issue number https://github.com/kootenpv/yagmail/commit/a2ea9e869f3bfd89af880e0755e19850971963d7

**Comment by MrCreosote** [comment]
Hi Pascal,

Turns out the fix was simple - I just needed to change `<myemail>@gmail.com` in the `SMTP()` call to simply `<myemail>`. 

I'm guessing yagmail assumes that the account is a gmail account when oauth2 is specified and maybe the code automatically appends `@gmail.com` to the username at some point, so my username wound up as `<myemail>@gmail.com@gmail.com`? Google wouldn't be happy about that, I presume.

Assuming something like that is going on, maybe the code could detect if `@gmail.com` is already appended and not append again if so?

Thanks for the help, yagmail is pretty darn spiffy now I've got it working.

**Comment by kootenpv** [comment]
I followed back the trace, and it seems something is going wrong? It shouldn't result in double @gmail.com. It only appens if there is no @ in the username: https://github.com/kootenpv/yagmail/blob/master/yagmail/sender.py#L335

Glad you got it working.

**Comment by MrCreosote** [comment]
doh, I see the issue: 
```
yagmail.SMTP('[redacted]@gmailcom', ...
                                                     ^
```

Hard to notice that if you're just using the history, and the error Google returns isn't very helpful...

Thanks again for the help.

**Comment by kootenpv** [comment]
I agree that the errors returned are not helpful at all!

Ironically, there's a "soft" check for obviously wrong emails, but it is only applied to the target "to" email address, not the "from" address. I guess I should also validate that one :)

**Comment by kootenpv** [comment]
I added the default email checking to also check whether a dot is found in email (excluding localhost). 
 
Available as of `yagmail-0.9.175`
